store/verkle/proof.go

```diff
@@ -73,29 +84,32 @@ func (p *ProofOp) Run(args [][]byte) ([][]byte, error) {
 }
 
 func (p *ProofOp) GetKey() []byte {
-	return p.Key
+	return p.Proof.Key
 }
 
-func (p *ProofOp) ProofOp() tmmerkle.ProofOp {
+func (p *ProofOp) ProofOp() crypto.ProofOp {
 	var data bytes.Buffer
 	enc := gob.NewEncoder(&data)
-	err := enc.Encode(p)
+	err := enc.Encode(p.Proof)
 	if err != nil {
-		return tmmerkle.ProofOp{}
+		return crypto.ProofOp{}
 	}
-	return tmmerkle.ProofOp{
+	return crypto.ProofOp{
 		Type: ProofType,
-		Key:  p.Key,
+		Key:  p.GetKey(),
 		Data: data.Bytes(),
 	}
 }
 
-func ProofDecoder(pop tmmerkle.ProofOp) (merkle.ProofOperator, error) {
+func ProofDecoder(pop crypto.ProofOp, root verkle.VerkleNode) (merkle.ProofOperator, error) {
```

Comment:
Func signature diff from `v2/smt` store. I think this is internal to us, so should be okay to do?